test/test_hydrostatic_free_surface_models.jl

```diff
@@ -0,0 +1,120 @@
+using Oceananigans: CPU, GPU
+using Oceananigans.Models: HydrostaticFreeSurfaceModel
+using Oceananigans.Grids: Periodic, Bounded
+
+function time_stepping_hydrostatic_free_surface_model_works(arch, topo, coriolis)
+    grid = RegularCartesianGrid(size=(1, 1, 1), extent=(2π, 2π, 2π), topology=topo)
+    model = HydrostaticFreeSurfaceModel(grid=grid, architecture=arch, coriolis=coriolis)
+    simulation = Simulation(model, Δt=1.0, stop_iteration=1)
+    run!(simulation)
+
+    return model.clock.iteration == 1
+end
+
+function hydrostatic_free_surface_model_tracers_and_forcings_work(arch)
+    grid = RegularCartesianGrid(size=(1, 1, 1), extent=(2π, 2π, 2π))
+    model = HydrostaticFreeSurfaceModel(grid=grid, architecture=arch, tracers=(:T, :S, :c, :d))
+    set!(model, η=1)
```

Comment:
Ah I guess we'll have to do this for a lot of tests, similar to `ShallowWaterModel`, to avoid blowup when timestepping due to division by zero?

Reply:
You mean setting the displacement to some constant value?

Since we're modeling displacement rather than the total thickness of the layer, I think we should get away with an initial value of 0. I can remove the call to `set!` since it's tested in another routine.

Reply:
Ah right it's testing `set!`.